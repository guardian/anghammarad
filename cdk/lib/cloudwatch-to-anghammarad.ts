import type { GuStackProps } from "@guardian/cdk/lib/constructs/core";
import {GuAnghammaradTopicParameter, GuStack} from "@guardian/cdk/lib/constructs/core";
import {GuLambdaFunction} from "@guardian/cdk/lib/constructs/lambda";
import {GuardianAwsAccounts} from "@guardian/private-infrastructure-config";
import type { App } from "aws-cdk-lib";
import {AnyPrincipal} from "aws-cdk-lib/aws-iam";
import {Runtime} from "aws-cdk-lib/aws-lambda";
import {SnsEventSource} from "aws-cdk-lib/aws-lambda-event-sources";
import {Topic} from "aws-cdk-lib/aws-sns";

export class CloudwatchToAnghammarad extends GuStack {
  constructor(scope: App, id: string, props: GuStackProps) {
    super(scope, id, props);
    const topic = new Topic(this, "CloudWatchToAnghammaradTopic", {
      topicName: "cloudwatch-to-anghammarad"
    });
    const accountNumbers = Object.values(GuardianAwsAccounts) as string[];
    accountNumbers.map((accountNumber) => {
      // https://docs.aws.amazon.com/sns/latest/dg/sns-access-policy-use-cases.html#sns-allow-cloudwatch-alarm-to-publish-to-topic-in-another-account
      const principal = new AnyPrincipal()
          .withConditions({
            'ArnLike': {
              'aws:SourceArn': `arn:aws:cloudwatch:${this.region}:${accountNumber}:alarm:*`
            }
          });
      topic.grantPublish(principal);
    });


    const anghammaradTopicArn = GuAnghammaradTopicParameter.getInstance(this).valueAsString;
    const lambda = new GuLambdaFunction(this, "CloudWatchToAnghammaradLambda", {
      app: "cloudwatch-to-anghammarad",
      environment: {
        ANGHAMMARAD_SNS_TOPIC_ARN: anghammaradTopicArn,
      },
      fileName: "cloudwatch.jar",
      handler: "com.gu.cloudwatch.Lambda::handleRequest",
      runtime: Runtime.JAVA_11
    });
    lambda.addEventSource(new SnsEventSource(topic));

    const anghammaradTopic = Topic.fromTopicArn(
        this,
        'AnghammaradTopic',
        anghammaradTopicArn,
    );

    anghammaradTopic.grantPublish(lambda);

  }
}
