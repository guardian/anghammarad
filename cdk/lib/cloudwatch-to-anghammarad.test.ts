import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { CloudwatchToAnghammarad } from "./cloudwatch-to-anghammarad";

describe("The CloudwatchToAnghammarad stack", () => {
  it("matches the snapshot", () => {
    const app = new App();
    const stack = new CloudwatchToAnghammarad(app, "CloudwatchToAnghammarad", { stack: "deploy", stage: "TEST" });
    const template = Template.fromStack(stack);
    expect(template.toJSON()).toMatchSnapshot();
  });
});
