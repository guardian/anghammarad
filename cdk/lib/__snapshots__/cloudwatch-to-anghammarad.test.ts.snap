// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CloudwatchToAnghammarad stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuAnghammaradTopicParameter",
      "GuDistributionBucketParameter",
      "GuLambdaFunction",
    ],
    "gu:cdk:version": "TEST",
  },
  "Parameters": {
    "AnghammaradSnsArn": {
      "Default": "/account/services/anghammarad.topic.arn",
      "Description": "SSM parameter containing the ARN of the Anghammarad SNS topic",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "DistributionBucketName": {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CloudWatchToAnghammaradLambda0A9908CC": {
      "DependsOn": [
        "CloudWatchToAnghammaradLambdaServiceRoleDefaultPolicyE691A130",
        "CloudWatchToAnghammaradLambdaServiceRole9B5AB4E8",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Ref": "DistributionBucketName",
          },
          "S3Key": "deploy/TEST/cloudwatch-to-anghammarad/cloudwatch.jar",
        },
        "Environment": {
          "Variables": {
            "ANGHAMMARAD_SNS_TOPIC_ARN": {
              "Ref": "AnghammaradSnsArn",
            },
            "APP": "cloudwatch-to-anghammarad",
            "STACK": "deploy",
            "STAGE": "TEST",
          },
        },
        "Handler": "com.gu.cloudwatch.Lambda::handleRequest",
        "MemorySize": 1024,
        "Role": {
          "Fn::GetAtt": [
            "CloudWatchToAnghammaradLambdaServiceRole9B5AB4E8",
            "Arn",
          ],
        },
        "Runtime": "java11",
        "Tags": [
          {
            "Key": "App",
            "Value": "cloudwatch-to-anghammarad",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/anghammarad",
          },
          {
            "Key": "Stack",
            "Value": "deploy",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "Timeout": 30,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CloudWatchToAnghammaradLambdaAllowInvokeCloudwatchToAnghammaradCloudWatchToAnghammaradTopicADB45B11912EFA70": {
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "CloudWatchToAnghammaradLambda0A9908CC",
            "Arn",
          ],
        },
        "Principal": "sns.amazonaws.com",
        "SourceArn": {
          "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "CloudWatchToAnghammaradLambdaCloudWatchToAnghammaradTopicF310CDEA": {
      "Properties": {
        "Endpoint": {
          "Fn::GetAtt": [
            "CloudWatchToAnghammaradLambda0A9908CC",
            "Arn",
          ],
        },
        "Protocol": "lambda",
        "TopicArn": {
          "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
        },
      },
      "Type": "AWS::SNS::Subscription",
    },
    "CloudWatchToAnghammaradLambdaServiceRole9B5AB4E8": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "cloudwatch-to-anghammarad",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/anghammarad",
          },
          {
            "Key": "Stack",
            "Value": "deploy",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "CloudWatchToAnghammaradLambdaServiceRoleDefaultPolicyE691A130": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                    ],
                  ],
                },
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":s3:::",
                      {
                        "Ref": "DistributionBucketName",
                      },
                      "/deploy/TEST/cloudwatch-to-anghammarad/cloudwatch.jar",
                    ],
                  ],
                },
              ],
            },
            {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/deploy/cloudwatch-to-anghammarad",
                  ],
                ],
              },
            },
            {
              "Action": [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/deploy/cloudwatch-to-anghammarad/*",
                  ],
                ],
              },
            },
            {
              "Action": "sns:Publish",
              "Effect": "Allow",
              "Resource": {
                "Ref": "AnghammaradSnsArn",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "CloudWatchToAnghammaradLambdaServiceRoleDefaultPolicyE691A130",
        "Roles": [
          {
            "Ref": "CloudWatchToAnghammaradLambdaServiceRole9B5AB4E8",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "CloudWatchToAnghammaradTopic058DA9BC": {
      "Properties": {
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/anghammarad",
          },
          {
            "Key": "Stack",
            "Value": "deploy",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "TopicName": "cloudwatch-to-anghammarad",
      },
      "Type": "AWS::SNS::Topic",
    },
    "CloudWatchToAnghammaradTopicPolicyC3E1475D": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000000:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "0",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000001:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "1",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000002:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "2",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000003:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "3",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":00000000004:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "4",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000005:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "5",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000006:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "6",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000007:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "7",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000008:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "8",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000009:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "9",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000010:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "10",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000011:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "11",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000012:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "12",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000013:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "13",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000014:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "14",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000015:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "15",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000016:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "16",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000017:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "17",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000018:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "18",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000019:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "19",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000020:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "20",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000021:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "21",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000022:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "22",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000023:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "23",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000024:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "24",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000025:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "25",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000026:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "26",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000027:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "27",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000028:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "28",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000029:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "29",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000030:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "30",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000031:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "31",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000032:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "32",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000033:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "33",
            },
            {
              "Action": "sns:Publish",
              "Condition": {
                "ArnLike": {
                  "aws:SourceArn": {
                    "Fn::Join": [
                      "",
                      [
                        "arn:aws:cloudwatch:",
                        {
                          "Ref": "AWS::Region",
                        },
                        ":000000000034:alarm:*",
                      ],
                    ],
                  },
                },
              },
              "Effect": "Allow",
              "Principal": {
                "AWS": "*",
              },
              "Resource": {
                "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
              },
              "Sid": "34",
            },
          ],
          "Version": "2012-10-17",
        },
        "Topics": [
          {
            "Ref": "CloudWatchToAnghammaradTopic058DA9BC",
          },
        ],
      },
      "Type": "AWS::SNS::TopicPolicy",
    },
  },
}
`;
